pepper-apis/src/main/java/org/broadinstitute/ddp/db/ActivityInstanceDao.java

```diff
@@ -378,35 +380,60 @@ public Long getActivityInstanceIdByGuid(Handle handle, String guid) {
     }
 
     /**
-     * Iterate through activity instance summaries and render the summary texts.
+     * Iterate through activity instance summaries and render the naming details and summary texts, as necessary. For
+     * activity instance name, the first instance may leverage content substitutions to render in it's activity number
+     * (which is 1). For activity instance with number greater than 1, if there is a "second name" then the second name
+     * will be used to render the name, otherwise " #N" will be appended to the original name (where N is the instance
+     * number).
      *
-     * @param handle the database handle
-     * @param userId the user who owns the activity instances
+     * @param handle    the database handle
+     * @param userId    the user who owns the activity instances
      * @param summaries the list of summaries
      */
-    public void renderActivitySummaryText(Handle handle, long userId, List<ActivityInstanceSummary> summaries) {
-        if (!summaries.isEmpty()) {
-            Set<Long> instanceIds = summaries.stream().map(ActivityInstanceSummary::getActivityInstanceId).collect(Collectors.toSet());
-            var instanceDao = handle.attach(org.broadinstitute.ddp.db.dao.ActivityInstanceDao.class);
-            var substitutions = instanceDao.bulkFindSubstitutions(instanceIds)
-                    .collect(Collectors.toMap(wrapper -> wrapper.getActivityInstanceId(), wrapper -> wrapper.unwrap()));
-            var sharedSnapshot = I18nContentRenderer
-                    .newValueProviderBuilder(handle, userId)
-                    .build().getSnapshot();
-            for (var summary : summaries) {
-                if (StringUtils.isBlank(summary.getActivitySummary())) {
-                    continue;
+    public void renderActivitySummary(Handle handle, long userId, List<ActivityInstanceSummary> summaries) {
```

Comment:
This method has the most significant changes. The rest of the PR is plumbing to pass through the new configuration data.